core/src/validator.rs

```diff
@@ -344,8 +344,29 @@ impl Validator {
                 .set_entrypoint(entrypoint_info.clone());
         }
 
-        // If the node was loaded from a snapshot, advertise it in gossip
         if let Some(snapshot_hash) = snapshot_hash {
+            if let Some(ref trusted_validators) =
+                config.snapshot_config.as_ref().unwrap().trusted_validators
+            {
+                let trusted = cluster_info
+                    .read()
+                    .unwrap()
+                    .get_snapshot_hash(snapshot_hash.0)
+                    .iter()
+                    .any(|(pubkey, hash)| {
+                        trusted_validators.contains(pubkey) && snapshot_hash.1 == *hash
+                    });
+
+                if !trusted {
```

Comment:
I went for a single attempt then hard fail here rather than polling gossip for a time.  `validator/src/main.rs` is only supposed to select a trusted snapshot to download

Reply:
ok; if it works. I was noticing when I ran the local_cluster tests specifically that the values didn't seem like they are propagated on the first attempt. That's why I had it doing the polling with a bit of waiting.

Reply:
Ah, I see.  I'll try that tomorrow and see what happens